src/MatrixHttpError.ts

```diff
@@ -0,0 +1,11 @@
+export interface MatrixHttpError {
```

Comment:
Going to be a pain again but I will say this is looking much better!

Could this become a class that extends Error so it ends up looking a bit like https://github.com/turt2live/matrix-bot-sdk/blob/main/src/models/MatrixError.ts?

This allows us to throw it without JS getting upset that we're throwing non-errors. It also means you can safely do away with `isMatrixHttpError` and instead do `err instanceof MatrixHttpError`

Reply:
yes!